### PR TITLE
OOP School Library: Decorate a class

### PR DESCRIPTION
**Pull request by DJ-MrJay**
- Created a class <code>Nameable</code>
- Turned <code>Person</code> class to <code>Nameable</code>
- Prepared base Decorator
- Prepared CapitalizeDecorator and TrimmerDecorator

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.slice(0, 9)
+  end
+end
```
Comment (omarsalem7) [inline nameable.rb:30]: - [ ] [OPTIONAL] It will be better if you separate each class in a separate file.

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @DJ-MrJay 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- The Pull Request has a proper title and description. ✔️
- Correct git-flow ✔️

## Required Changes ♻️

- [ ] Kindly consider the `Person` class inherits from `Nameable` class. It also has a method `correct_name`.

## Optional suggestions
_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 2]
```diff
@@ -1,8 +1,11 @@
-class Person
+require_relative './nameable'
+
+class Person < Nameable
   attr_accessor :name, :age
   attr_reader :id
```
Comment (addod19) [inline person.rb:6]: - [ ] It would be better if we create a `correct_name` method for our `Person` class as mentioned by the previous code reviewer

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @DJ-MrJay, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [ ] Git-flow followed :+1: 
- [ ] README file is descriptive :rocket: 
- [ ] Great job addressing some of the comments :100: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — APPROVED** [review]
Hi @DJ-MrJay  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

littlebigwhale-lbw


## Highlights
All linters are passing :heavy_check_mark: 
Great effort in making the requested changes :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zieeco — COMMENTED** [review]
Hi @DJ-MrJay,

According to this [review summary](https://github.com/DJ-MrJay/oop-school-library/pull/2#pullrequestreview-1238740596), Your project has been approved.

Cheers and Happy Coding!

**Pull request merged**